### ISSUE TITLE
[1.7.1] Containers can be created with an invalid restart policy name

### ISSUE THREAD

**Issue by sanmai-NL** [docker/compose#3481]
`docker-compose.yml`:

``` yaml
version: '2'
services:
  alpine:
    image: "alpine:latest"
    restart: unlesss-stopped
```

In the previous, `unlesss-stopped` is an invalid value for the restart policy name.

``` sh
sudo docker-compose up
```

Does not complain, and a container is created and started with an incorrect restart policy. Incorrect, based on `docker inspect` output.

 In contrast, this fails:

``` sh
sudo docker create --restart='unlesss-stopped' alpine
```

```
docker: invalid restart policy unlesss-stopped.
See 'docker create --help'.
```

**Context**

```
docker-compose version 1.7.1, build 6c29830
Docker version 1.11.1, build 5604cbe
```

**Comment by dnephin** [comment]
This is a bug in docker engine. The engine is responsible for providing an error if it doesn't understand the restart policy it receives.

The validation is being done client side in the docker cli, which is wrong.

**Comment by johnharris85** [comment]
This is fixed now that [#24073](https://github.com/docker/docker/pull/24073) has been merged into the engine.
